### PR TITLE
[ADD] module attachment_file_extension

### PR DESCRIPTION
**Pull request by tafaRU**
Please refer to README file for further info.

### CODE REVIEW

File: attachment_file_extension/models/ir_attachment.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        for attachment in self.browse(cr, uid, ids, context=context):
+            if attachment.extension:
+                with self._file_extension_lock:
+                    self._extension = attachment.extension
+                    super(IrAttachment, self).unlink(
+                        cr, uid, ids, context=context)
+            else:
+                super(IrAttachment, self).unlink(cr, uid, ids, context=context)
+
+    _columns = {
```
Comment (hbrunn) [inline attachment_file_extension/models/ir_attachment.py:80]: why old style declarations?

File: attachment_file_extension/models/ir_attachment.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import os.path
+import threading
+
+
+class IrAttachment(orm.Model):
+    _inherit = 'ir.attachment'
+
+    def __init__(self, *args, **kwargs):
+        self._file_extension_lock = threading.RLock()
+        self._extension = None
```
Comment (hbrunn) [inline attachment_file_extension/models/ir_attachment.py:32]: please explain this

File: attachment_file_extension/models/ir_attachment.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                    super(IrAttachment, self).unlink(
+                        cr, uid, ids, context=context)
+            else:
+                super(IrAttachment, self).unlink(cr, uid, ids, context=context)
+
+    _columns = {
+        'datas': fields.function(
+            _data_get, fnct_inv=_data_set, string='File Content',
+            type="binary", nodrop=True),
+        'extension': fields.char(string='Attachment File Extension', size=4),
```
Comment (hbrunn) [inline attachment_file_extension/models/ir_attachment.py:84]: i wouldn't restrict the size here. extensions can be arbitrarily long

**Comment by hbrunn** [comment]
I must admit I don't really understand this module. Why wouldn't you leave the filename alone and make extension a function field?

**Comment by hbrunn** [comment]
and in any case, you should provide an init hook to calculate the extension for existing attachments

**Comment by hbrunn** [comment]
closing due to inactivity

**Comment by tafaRU** [comment]
@hbrunn, I apologise for not giving updates, please consider to re-open this PR.
I'll try to allocate some time to address your remarks.

Thank you in advance.

**Comment by tafaRU** [comment]
If you cannot, I'll open a new one, let me know.

Thanks.

**Comment by tafaRU** [comment]
Thank you, @hbrunn

**Comment by StefanRijnhart** [comment]
@tafaRU now that the PR has been reopened, could you follow up?

**Comment by tafaRU** [comment]
@StefanRijnhart, sorry I'm quite busy actually, sure I will take in consideration @hbrunn's remarks but I cannot promise any deadline.

**Comment by hbrunn** [comment]
closing due to inactivity

**Pull request closed without merging**